lib/active_resource/singleton.rb

```diff
@@ -107,5 +130,9 @@ def create
       def singleton_path(options = nil)
         self.class.singleton_path(options || prefix_options)
       end
+
+      def custom_method_element_url(method_name, options = {})
+        "#{self.class.prefix(prefix_options)}#{self.class.collection_name}/#{method_name}#{self.class.format_extension}#{self.class.__send__(:query_string, options)}"
+      end
```

Comment:
This implementation is copied directly from `ActiveResource::CustomMethods`, then altered to omit the `id` portion of the path:

https://github.com/rails/activeresource/blob/9c8a2ee30325ff0bf789a239cc25fa361ec4a887/lib/active_resource/custom_methods.rb#L121-L123

```diff
 def custom_method_element_url(method_name, options = {})
-  "#{self.class.prefix(prefix_options)}#{self.class.collection_name}/#{URI.encode_www_form_component(id.to_s)}/#{method_name}#{self.class.format_extension}#{self.class.__send__(:query_string, options)}"
+  "#{self.class.prefix(prefix_options)}#{self.class.collection_name}/#{method_name}#{self.class.format_extension}#{self.class.__send__(:query_string, options)}"
 end
```